helix-term/src/commands/lsp.rs

```diff
@@ -182,13 +186,24 @@ impl ui::menu::Item for PickerDiagnostic {
     }
 }
 
-fn location_to_file_location(location: &lsp::Location) -> FileLocation {
-    let path = location.uri.to_file_path().unwrap();
-    let line = Some((
-        location.range.start.line as usize,
-        location.range.end.line as usize,
-    ));
-    (path.into(), line)
+/// Creates a [`FileLocation`] from an [`lsp::Location`], for use with picker previews.
+///
+/// This will return [`None`] if the location uri's scheme is not "file".
+fn location_to_file_location(location: &lsp::Location) -> Option<FileLocation> {
+    match uri_to_file_path(&location.uri) {
+        Ok(path) => {
+            let line = Some((
+                location.range.start.line as usize,
+                location.range.end.line as usize,
+            ));
+            Some((path.into(), line))
+        }
+        Err(FilePathError::UnsupportedScheme) => None,
+        Err(FilePathError::UnableToConvert) => unreachable!(
+            "file path should be able to be acquired from URI: {}",
+            location.uri
+        ),
+    }
```

Comment:
I don't really think the panic is correct here this actually caused crashes in the past. The real fix is #7367 not sure what to do with this until then.

Reply:
i've not to panic here to air on the safe side, so both simply return `None`. note that if this crash is pervasive enough, #7367 isn't a complete fix since this function is also used for goto defintion and friends.